datalad/crawler/main.py

```diff
@@ -232,7 +233,8 @@ def page2annex(self, existing='check', dry_run=False, cache=False):
                         incoming_filename, incoming_downloaded, incoming_updated, downloaded_size = \
                           download_url_to_incoming(href_full, incoming_annex.path,
                                        join(repo_sectiondir, href_dir),
-                                       db_incoming=db_incoming, dry_run=self.runner.dry,  # TODO -- use runner?
+                                       db_incoming=db_incoming,
+                                       dry_run=self.runner.protocol.__class__.__name__ == 'DryRunProtocol',  # TODO -- use runner?
```

Comment:
and anyways you are creating a protocol based on what dry_run arg is -- so why to complicate

side-note: I feel like protocol's should become some kind of a decorator, may be indeed not a context manager, around function calls etc  and here (and below in page2annex where use of dry_run is maintained)... but later